addons/binding/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/thermostat/NATherm1Handler.java

```diff
@@ -89,23 +92,100 @@ protected State getNAThingProperty(String channelId) {
+                    NAZone zone = getZone(currentProgram.getZones(), currentProgramMode.getId());
+                    return toStringType(zone.getName());
+                }
+            }
+        }
+        return UnDefType.UNDEF;
+    }
+
+    private State getCurrentSetpoint() {
+        if (module != null && module.getSetpoint() != null) {
+            NASetpoint setPoint = module.getSetpoint();
+            String currentMode = setPoint.getSetpointMode();
+
+            NAThermProgram currentProgram = module.getThermProgramList().stream().filter(p -> p.getSelected() != null)
+                    .findFirst().get();
+            if (currentProgram != null) {
+                switch (currentMode) {
+                    case CHANNEL_SETPOINT_MODE_MANUAL:
+                        return toDecimalType(setPoint.getSetpointTemp());
+                    case CHANNEL_SETPOINT_MODE_AWAY:
+                        NAZone zone = getZone(currentProgram.getZones(), 2);
+                        return toDecimalType(zone.getTemp());
+                    case CHANNEL_SETPOINT_MODE_HG:
+                        NAZone zone1 = getZone(currentProgram.getZones(), 3);
+                        return toDecimalType(zone1.getTemp());
+                    case CHANNEL_SETPOINT_MODE_PROGRAM:
+                        NATimeTableItem currentProgramMode = getCurrentProgramMode(module.getThermProgramList());
+                        if (currentProgramMode != null) {
+                            NAZone zone2 = getZone(currentProgram.getZones(), currentProgramMode.getId());
+                            return toDecimalType(zone2.getTemp());
+                        }
+                    case CHANNEL_SETPOINT_MODE_OFF:
+                    case CHANNEL_SETPOINT_MODE_MAX:
+                        return UnDefType.NULL;
+                }
+            }
+        }
+        return UnDefType.UNDEF;
+    }
+
+    private NAZone getZone(List<NAZone> zones, int searchedId) {
+        NAZone result = zones.stream().filter(z -> z.getId() == searchedId).findFirst().get();
+        return result;
+    }
+
+    private long getNetatmoProgramBaseTime() {
         Calendar mondayZero = Calendar.getInstance();
         mondayZero.set(Calendar.DAY_OF_WEEK, Calendar.MONDAY);
         mondayZero.set(Calendar.HOUR_OF_DAY, 0);
         mondayZero.set(Calendar.MINUTE, 0);
         mondayZero.set(Calendar.SECOND, 0);
 
+        return mondayZero.getTimeInMillis();
+    }
+
+    private NATimeTableItem getCurrentProgramMode(List<NAThermProgram> thermProgramList) {
+        NATimeTableItem lastProgram = null;
         Calendar now = Calendar.getInstance();
-        long diff = (now.getTimeInMillis() - mondayZero.getTimeInMillis()) / 1000 / 60;
+
+        long diff = (now.getTimeInMillis() - getNetatmoProgramBaseTime()) / 1000 / 60;
+
+        NAThermProgram currentProgram = thermProgramList.stream()
+                .filter(p -> p.getSelected() != null && p.getSelected()).findFirst().get();
```

Comment:
In https://github.com/openhab/openhab2-addons/pull/3097/files#diff-2f4f50ac7d19c5baba0c0102cd5fd4c2R108 and https://github.com/openhab/openhab2-addons/pull/3097/files#diff-2f4f50ac7d19c5baba0c0102cd5fd4c2R126 you used `p -> p.getSelected() != null` for filter condition. You added a condition (`p.getSelected()`) here. Why?